### DISCUSSION TITLE
How to update to Nextcloud 28 now?

### DISCUSSION THREAD

**Discussion by szaimen** [nextcloud/all-in-one#2692] [Wiki]
If you don't want to wait for AIO getting updated to NC28 for everyone in the near future, you can upgrade to Nextcloud 28 already now by following these steps:

1. First, open the aio interface and create a backup and start your containers again after the backup is successful.

2. Afterwards, run via CLI:
```
# Go into the container
sudo docker exec --user www-data -it nextcloud-aio-nextcloud bash
```

3. Now inside the container:

```
# Switch to the beta channel
php occ config:system:set updater.release.channel --value=beta

# Run the update
php updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force

# Switch back to the stable channel
php occ config:system:set updater.release.channel --value=stable

# Exit the container
exit
```

The above instructions should upgrade you to Nextcloud 28. You might need to follow the steps a second time to reach NC28. If something goes wrong, feel free to restore the backup.

**Comment by martin-77** [comment]
UPDATE 1:
Error was caused by App "Bookmark" - disabled and seems to fix the Error...

ran an update on 7.8.1 - 27.1.4 to 28
Error in integrity-check:

Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- INVALID_HASH
		- core/js/mimetypelist.js

Raw output
==========
Array
(
    [core] => Array
        (
            [INVALID_HASH] => Array
                (
                    [core/js/mimetypelist.js] => Array
                        (
                            [expected] => 550ab566d30693bfa24ec4b15d9df87731ae8a3be8f79dabf94757e5b8b20eec6e4b678f17af1718297f2872f6b04519eeb024d1dff11947f29da431c7f11201
                            [current] => 24e6faa4a1361e51ff70027024f41c4b37fdb5f6f347ceb095450ddbde2d0a6b6dd1403be92a912d2889064a9736a9bed89d7dc62470035fc1299d4b977c5f46
                        )

                )

        )

)

**Comment by caloracloud** [comment]
Unfortunately, it did not work for me !!

**Reply by szaimen** [reply]
In that case you can restore a backup.

**Comment by rccerrillo** [comment]
I was successful in updating but it did not complete the indexing of my db.  I will just wait for the final release.

**Comment by Zakmaf** [comment]
Worked fine for me.
I just had to run the update twice until reaching version 28.

**Comment by jospoortvliet** [comment]
Note that next week a first minor update comes out with a bunch of little UI fixes and other fixes. Some things like being able to move files with the MOVE button in the UI got lost, those will be restored... I think there's also a bulk-move feature that should make moving files faster. I don't think there's anything huge but it is all nice to have.

**Comment by Abstracting** [comment]
I had to run the update command 2 times. Then it worked:
`php updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force
`

**Comment by elchmartin** [comment]
for me it worked, first updated from 27.1.4 to 27.1.5, running the updater a second time brought me to 28.0.1 RC1.

**Reply by auceto** [reply]
Sorry Sir, but what do you mean "running the updater a secon time"?
that I've to repeat the above instruction or whatelse
For me nothing happen with the above instruction my release stuck on 27
thanks

**Reply by elchmartin** [reply]
Run this two times:

# Run the update php updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force

Did the Trick for me

**Reply by jpnilson** [reply]
When I followed the instructions twice I got to the version 29 beta.  After doing a restore I ran through the instructions switching to beta the first time. The second time I only ran the update command while in the stable release mode.  That brought me to version 28.0.3

**Comment by JesseKPhillips** [comment]
What would be the release timeline for aio?

**Comment by elephantastyczny** [comment]
Good further here with @JesseKPhillips question.

I did migrate to NC 28.0.0 manually but I'm wondering if I can update to NC 28.0.1 safely?

-----------
Update
A new version is available: Nextcloud 28.0.1
1 app has no compatible version for this Nextcloud version available.

Apps missing compatible version 

[Nextcloud All-in-One ↗](https://apps.nextcloud.com/apps/nextcloud-aio)

**Reply by martin-77** [reply]
Worked for me.without any issues. Now running 28.0.1

**Reply by elephantastyczny** [reply]
Thx!

**Reply by auceto** [reply]
Sorry but the link above not work

**Reply by Shinebayart** [reply]
how to migrate 28.0.0 to 28.0.1 ?

**Reply by elephantastyczny** [reply]
[#2692](https://github.com/nextcloud/all-in-one/discussions/2692)

Go to container and then run the update :)

**Comment by corndog2000** [comment]
This doesn't seem to work with the Unraid docker image. I get this error on the second step.

```
/var/www/docker-aio # php occ config:system:set updater.release.channel --value=beta
Could not open input file: occ
```

**Reply by elephantastyczny** [reply]
You have to run it from a context of www-data user

**Reply by elephantastyczny** [reply]
sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ …

**Reply by corndog2000** [reply]
I still have the same issue.

```
root@unraid:~# sudo docker exec --user www-data -it nextcloud-aio-mastercontainer php occ config:system:set updater.release.channel --value=beta
Could not open input file: occ
root@unraid:~# 
```

**Reply by corndog2000** [reply]
Ah apologies. I was running in the mastercontainer and not the nextcloud container.

**Reply by KlomDark** [reply]
I think the error is just complaining that it can't find `occ`.

Try putting `/var/www/html/nextcloud/` (or wherever you have NC installed) in front of `occ`

Ex: `/var/www/html/nextcloud/occ`

**Comment by CrackShotDave** [comment]
Worked perfectly, thankyou very easy to follow instructions as always!!

I now have 2 warnings however, I'm sure they are simple fixes but having searched I am at a loss..

1) Server has no maintenance window start time configured. This means resource intensive daily background jobs will also be executed during your main usage time. We recommend to set it to a time of low usage, so users are less impacted by the load caused from these heavy tasks. 

2) The database is missing some indexes. Due to the fact that adding indexes on big tables could take some time they were not added automatically. By running "occ db:add-missing-indices" those missing indexes could be added manually while the instance keeps running. Once the indexes are added queries to those tables are usually much faster. Missing optional index "fs_storage_path_prefix" in table "filecache".

**Reply by auceto** [reply]
Same here, I done restore because I dont know how to fix it

**Reply by CrackShotDave** [reply]
OK so clearly you have to run the "occ db:add-missing-indices" command, im not sure how to do this tho... i have read all the guides 100 times!!

**Reply by CrackShotDave** [reply]
Success!!! i looked at the post above - different issue - and used that command:
sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ db:add-missing-indices

Now just trying to figure out the maintenance window stuf

**Reply by auceto** [reply]
For fixing the maintenance window stuf I found this in NC's support
[https://help.nextcloud.com/t/server-has-no-maintenance-window-start-time-configured/180480/13]

In summary you need to execute this command

sudo -u www-data php /path/to/nextcloud/occ config:system:set maintenance_window_start --value="1" --type=integer`

to add the new maintenance_window_start --value="1" parameter in config.php file

I haven't tried it yet

**Comment by martin-77** [comment]
To fix the missing indexes simply run sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ db:add-missing-indices.

**Comment by h9419** [comment]
I want to add a small tip to everyone seeing this since this issue is linked in the Nextcloud AIO interface. 

If you see `Warning: Failed to set memory limit to 0 bytes` and upgrade fails, first delete the `.step` file shown created by the updater located in `/nc-updater/updater-xxxxxxxxxxxx/.step`. Then try again with the following command

```
sudo -u www-data php -d memory_limit=2G updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force
```

My default installation of Nextcloud AIO does not modify the default PHP memory limit so running the updater as is will result in out of memory exception during extract and verify.

**Comment by xyhhx** [comment]
I'm using the [standalone image](https://github.com/nextcloud/docker) at version 28.0.2. Can I migrate to AIO by deploying it at version 27, running the upgrade procedure documented here, then following the [migration steps here](https://github.com/nextcloud/all-in-one/blob/main/migration.md#migrate-the-files-and-the-database) (from step 3 onward)?

**Reply by jstebbins** [reply]
I did this a couple of weeks ago. Start at Step 1 and read everything carefully. Step 5 in the migration instructions is where you would install AIO and run this upgrade procedure. It worked for me, but given the complexity, I'm not going to make any guarantees. Definitely make sure you have all the backups necessary to get back to where you started.

**Comment by vijay-prema** [comment]
**EDIT: Now that version 28 is no longer in beta, you should NOT run the commands to "Switch to the beta channel" because you will end up at the new beta v29.  Either way, pay attention to the console message that will tell you which version it will move to and ask for confirmation**

I had 27.1.5 AIO docker, and successfully updated to 28.0.3.  As mentioned by others, I had to run the second command `php updater/updater.phar.... etc` TWICE.

If you see this message when running the updater command, then you will need to run it again right after it completes, to get to version 28
`Update to Nextcloud 27.1.7 RC1 available. (channel: "beta")`
So the second time running it should say instead:
`Update to Nextcloud 28.0.3 RC1 available. (channel: "beta")`

Next I noticed errors in the admin overview:
![image](https://github.com/nextcloud/all-in-one/assets/30308768/3703c80e-e518-4db2-92e3-9e188f4c9812)

- The integrity check error doesn't seem to be anything serious. It's just some files related to drawio plugin mime types, so I will leave it alone for now.  Maybe this will just fix itself next time the drawio plugin gets updated? _(EDIT: indeed this went away after updating to 28 final release recently)_
- For the `maintenance window` warning, I ran `php occ config:system:set maintenance_window_start --type=integer --value=13` inside the container, which sets the maintenance time to 1pm UTC (13:00).
- The last warning (database missing indexes) is fixed by running `php occ db:add-missing-indices` in the container
- I checked the Logging section of Admin and the 32 errors it refers to did not seem critical so I ignored them.  I rebooted the host system and no errors occurred after, so this warning message went away.

**Comment by punkyard** [comment]
Hi @szaimen 
can we see somewhere a roadmap and the future integration of NC28 in AiO?
no rush, just being curious
Thanks!

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one/issues/3888

**Reply by punkyard** [reply]
thanks @szaimen 
it's due tomorrow!!! ✌️
https://github.com/nextcloud/3rdparty/milestone/250

**Comment by vijay-prema** [comment]
@szaimen Perhaps your instructions should be changed to remove the "Switch to the beta channel" lines, now that 28 is no longer beta?  People following these instructions will end up on 29 otherwise.

**Reply by olehj** [reply]
yup.. I did accidentally. Now I can't update from .3 to .4 due to some bugs:

`
d62c3c871e75:/var/www/html$ php updater/updater.phar --no-interaction --no-backup && php occ app:enable nextcloud-aio --force
Nextcloud Updater - version:  dirty

PHP Fatal error:  Uncaught Error: Class "OC" not found in /var/www/html/config/apps.config.php:5
Stack trace:
#0 phar:///var/www/html/updater/updater.phar/lib/Updater.php(66): require_once()
#1 phar:///var/www/html/updater/updater.phar/lib/UpdateCommand.php(90): NC\Updater\Updater->__construct('/var/www/html/u...')
#2 phar:///var/www/html/updater/updater.phar/vendor/symfony/console/Command/Command.php(298): NC\Updater\UpdateCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#3 phar:///var/www/html/updater/updater.phar/vendor/symfony/console/Application.php(1040): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 phar:///var/www/html/updater/updater.phar/vendor/symfony/console/Application.php(301): Symfony\Component\Console\Application->doRunCommand(Object(NC\Updater\UpdateCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#5 phar:///var/www/html/updater/updater.phar/vendor/symfony/console/Application.php(171): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#6 phar:///var/www/html/updater/updater.phar/updater.php(10): Symfony\Component\Console\Application->run()
#7 /var/www/html/updater/updater.phar(14): require('phar:///var/www...')
#8 {main}
  thrown in /var/www/html/config/apps.config.php on line 5
d62c3c871e75:/var/www/html$
`

**Reply by Delor3an91** [reply]
@olehj ,

Same here...

**Reply by szaimen** [reply]
See https://github.com/nextcloud/server/issues/44157#issuecomment-1996703756

Alternatively feel free to restore the backup that you took before updating manually

**Reply by olehj** [reply]
Thanks, that worked!
